pepper-apis/src/main/java/org/broadinstitute/ddp/customexport/export/CustomExporter.java

```diff
@@ -0,0 +1,224 @@
+
+        String activityCode = activityDto.getActivityCode();
+        List<ActivityVersionDto> versionDtos = jdbiActivityVersion.findAllVersionsInAscendingOrder(activityDto.getActivityId());
+        for (ActivityVersionDto versionDto : versionDtos) {
+            // Only supports form activities for now.
+            FormActivityDef def = store.getActivityDef(studyGuid, activityCode, versionDto.getVersionTag());
+            if (def == null) {
+                def = formActivityDao.findDefByDtoAndVersion(activityDto, versionDto);
+                store.setActivityDef(studyGuid, activityCode, versionDto.getVersionTag(), def);
+            }
+            activities.add(new ActivityExtract(def, versionDto));
+        }
+
+        return activities;
+    }
+
+    /**
+     * Extracts all versions of custom export's activity
+     * @param handle the database handle
+     * @return list of extracts
+     */
+    public List<ActivityExtract> extractCustomActivity(Handle handle) {
+        JdbiActivity jdbiActivity = handle.attach(JdbiActivity.class);
+
+        Optional<ActivityDto> activityDtoOptional = jdbiActivity.findActivityByStudyGuidAndCode(customGuid, customActivity);
+        if (activityDtoOptional.isEmpty()) {
+            LOG.error("Custom enrollment activity DTO not found");
+            return null;
+        }
+        ActivityDto activityDto = activityDtoOptional.get();
+        List<ActivityExtract> activities = extractVersionsOfActivity(handle, activityDto, customGuid);
+
+        LOG.info("Custom export found {} versions of enrollment activity", activities.size());
+
+        return activities;
+    }
+
+    /**
+     * Export the given dataset to the given output using CSV formatting. Caller is responsible for closing the given output writer.
+     *
+     * @param activities   the list of activity data for a study, with additional data pre-computed
+     * @param participants the participant data for a study
+     * @param output       the output writer to use
+     * @return number of participant records written
+     * @throws IOException if error while writing
+     */
+    public int exportDataSetAsCsv(StudyDto studyDto, List<ActivityExtract> activities, Iterator<Participant> participants,
+                                  Writer output) throws IOException {
+        ParticipantMetadataFormatter participantMetaFmt;
+
+
+        List<String> excludedFields = new ArrayList<>(cfg.getStringList(CustomExportConfigFile.EXCLUDED_PARTICIPANT_FIELDS));
+        participantMetaFmt = new ParticipantMetadataFormatter(excludedFields);
+
+
+        List<String> headers = new LinkedList<>(participantMetaFmt.headers());
+
+        Map<String, Integer> activityTagToNormalizedMaxInstanceCounts = new HashMap<>();
+        Map<String, ActivityResponseCollector> responseCollectors = new HashMap<>();
+        Map<String, ActivityAttributesCollector> attributesCollectors = new HashMap<>();
```

Comment:
Since this is custom code, you can probably get rid of the whole business with `activityTagToNormalizedMaxInstanceCounts` and `attributesCollectors`. Those aren't really applicable to RGP.